rcldotnet/Clock.cs

```diff
@@ -0,0 +1,296 @@
+/* Copyright 2023 Queensland University of Technology.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using builtin_interfaces.msg;
+using ROS2.Utils;
+
+namespace ROS2
+{
+    internal static class TimeConstants
+    {
+        public const long NanosecondsPerSecond = 1000L * 1000L * 1000L;
+        public const long NanosecondsPerMillisecond = 1000000L;
+
+        public const long NanosecondsPerTimespanTick = NanosecondsPerMillisecond / TimeSpan.TicksPerMillisecond; // ~= 100.
+    }
+
+    /// <summary>
+    /// Time source type, used to indicate the source of a time measurement.
+    ///
+    /// ROSTime will report the latest value reported by a ROS time source, or
+    /// if a ROS time source is not active it reports the same as RCL_SYSTEM_TIME.
+    /// For more information about ROS time sources, refer to the design document:
+    /// http://design.ros2.org/articles/clock_and_time.html
+    ///
+    /// SystemTime reports the same value as the system clock.
+    ///
+    /// SteadyTime reports a value from a monotonically increasing clock.
+    /// </summary>
+    public enum ClockType
+    {
+        ROSTime = 1,
```

Comment:
Should be `RosTime` to be consistent with the `ClockChange` enum and the .NET naming guidelines.